### ISSUE TITLE
BlocksByRange under WS

### ISSUE THREAD

**Issue by twoeths** [ChainSafe/lodestar#2556]
**Is your feature request related to a problem? Please describe.**

+ Implement this https://github.com/ethereum/eth2.0-specs/pull/2131/files
+ Part of #2530
+ Since it's quite isolated I make it a separate issue to avoid pulling us from being compliant to the whole spec

**Detail**
1. Since we only need to serve blocks within `MIN_EPOCHS_FOR_BLOCK_REQUESTS`  epochs, we can prune our db accordingly in order to avoid having our db grow forever
2. Validate `beacon_blocks_by_range` request accordingly
3. We have to backfill blocks if starting from a ws checkpoint
+ Don't need to get through state transition, just validate block proposer and parent-child relationship until recent block roots inside the ws state
+ Consider hooking into / add new sync type  https://github.com/ChainSafe/lodestar/blob/074649e35fe6a20a5aeab8898b67b73dca2e2900/packages/lodestar/src/sync/range/range.ts#L112

**Comment by dapplion** [comment]
This new sync type can be actually fully independent from the other sync types. A node can be syncing forward at the same time it's backfilling.

I would create a new class BackfillSync that re-uses the `SyncChain` class which contains all the logic to download in parallel, load balance peers, etc. But with custom instantiation parameters

```ts
processChainSegment: (blocks) => {
  verifyParentChildSeg(blocks)
  db.blocks.put(blocks)
}
```

Also override the computeTarget function too to make it static and equal to the ws checkpoint

**Comment by g11tech** [comment]
some conversation  context:

![image](https://user-images.githubusercontent.com/76567250/135662917-021c55db-ce9a-4ca0-a8fb-4b333a63f09a.png)
![image](https://user-images.githubusercontent.com/76567250/135663009-0bb1b48b-e9c5-4e6c-9d20-d96c3389d9c4.png)
![image](https://user-images.githubusercontent.com/76567250/135663074-823dcdd2-d687-4973-9500-82fecc1072f8.png)
